src/porepy/models/solution_strategy.py

```diff
@@ -450,21 +460,17 @@ def after_nonlinear_convergence(
             values=solution, time_step_index=0, additive=False
         )
         self.convergence_status = True
-
+        self.nonlinear_solver_statistics.save()
```

Comment:
Perhaps this should be done as part of save_data_time_step()?

Reply:
Would be OK for me. However, I would then aim for a consistent change. This would include:
- Move the handling of the path where to store the performance data also to DataSavingMixin

In addition, I would advocate for saving the data also after failed convergence, as now, such that in the end, the routine would not entirely vanish.

Reply:
I agree. We could, though, consider calling `save_data_time_step` after failure as well.